protocols/grpc/agent.proto

```diff
@@ -434,3 +435,20 @@ message StringUser {
 	string gid = 2;
 	repeated string additionalGids = 3;
 }
+
+message CopyFileRequest {
+	// Path is the destination path in the guest.
+	string path = 1;
+	// FileSize is the expected file size, for security reasons write operations
+	// are made in a temporary file, once it has the expected size, it's moved
+	// to the destination path.
+	int64 file_size = 2;
+	// FileMode is the file mode.
+	uint32 file_mode = 3;
```

Comment:
Don't we also need to send `uid`+`gid` for both file and directory as currently they will only ever be created as the user the agent is running as (`root`)? Maybe less important but also what about `atime`, `mtime`, `ctime`?

Reply:
yes files are created as `root`, I'll add `uid` and `gid`, the other options can be added later on, what do you think?

Reply:
sounds good.